controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -18,26 +20,40 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+type GetUsageFunc func(ctx context.Context, cl runtimeclient.Client, clusterName string, toolchainStatusNs string) (*toolchainv1alpha1.ConsumedCapacity, error)
+
 // Reconciler is the reconciler for the SpaceProvisionerConfig CRs.
 type Reconciler struct {
 	Client runtimeclient.Client
+	// GetUsageFunc is a function that can be used in the tests to mock the fetching of the consumed usage. It defaults to collecting the actual usage
+	// from the ToolchainStatus.
+	GetUsageFunc GetUsageFunc
 }
 
 var _ reconcile.Reconciler = (*Reconciler)(nil)
 
-func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&toolchainv1alpha1.SpaceProvisionerConfig{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Watches(
 			&toolchainv1alpha1.ToolchainCluster{},
-			handler.EnqueueRequestsFromMapFunc(MapToolchainClusterToSpaceProvisionerConfigs(ctx, r.Client)),
+			handler.EnqueueRequestsFromMapFunc(MapToolchainClusterToSpaceProvisionerConfigs(r.Client)),
+		).
+		// we use the same information as the ToolchainStatus specific for the SPCs. Because memory consumption is
+		// read directly out of the member clusters using remote connections, let's look for it only once
+		// in ToolchainStatus and just read it out "locally" here without needing to reach out to the member clusters
+		// again.
```

Comment:
That's how I interpreted this line: https://github.com/codeready-toolchain/host-operator/blob/master/controllers/toolchainstatus/toolchainstatus_controller.go#L442.

Was I wrong?

Reply:
yaeh, it's reading the values from the MemberStatus, but it doesn't read the consumption form the NodeMetrics and Node resources, this is what I understood from the sentence: "Because memory consumption is read directly out of the member clusters", so I was afraid that we get the Node & NodeMetrics somewhere